src/main/java/com/artipie/asto/Key.java

```diff
@@ -142,6 +159,19 @@ public String string() {
             return String.join(From.DELIMITER, this.parts);
         }
 
+        @Override
+        public Optional<Key> parent() {
+            final Optional<Key> parent;
+            if (this.equals(Key.ROOT)) {
+                parent = Optional.empty();
+            } else {
+                parent = Optional.of(
+                    new Key.From(this.parts.subList(0, this.parts.size() - 1))
```

Comment:
@brastak `parts.size()` can be 0 and we can get `IndexOutOfBoundsException`. We should add additional check

Reply:
If parts.size() == 0, then the key should be equal to Key.ROOT. 
But now we have a problem with it, because of Key is interface and other implementation can exists. For example, new Key.Wrap(Key.ROOT) {} will fail with this check. I will fix the implementation